### DISCUSSION TITLE
Few questions on runner update -> 2.0.0

### DISCUSSION THREAD

**Discussion by aignacio** [cocotb/cocotb#4220] [Q&A]
Hey Cocotb dev team,

thanks for all the work that is coming up in 2.0.0 release, I just watched the [presentation](https://youtu.be/n7Ektuv-ONk?si=EsZNuUFayj5cKTb2&t=123) about the new features and w.r.t to the [runner](https://docs.cocotb.org/en/latest/runner.html), I have two questions:

* How is the support for Python coverage while using the runner? I tried to move to the runner approach in an [extension](https://github.com/aignacio/cocotbext-ahb) that I developed but the coverage reports were empty and I got a few warnings:
```bash
/test/.nox/run-3-12/lib/python3.12/site-packages/coverage/control.py:894: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")
...
```

I tried to specify the following passing as env but have had no luck 
```python
    session.env["COCOTB_COVERAGE_RCFILE"] = os.path.abspath(".coveragerc")
    session.env["COVERAGE_RC_FILE"] = os.path.abspath(".coveragerc")
    session.env["COCOTB_USER_COVERAGE"] = "1"
    session.env["COVERAGE"] = "1"
```
[Updated code to exemplify the issue.](https://github.com/aignacio/DELETE_ME/blob/master/tests/test_ahb.py#L128-L147)

-------

* While moving from cocotb-test to the runner, I had to update from installing my package as editable ([pip install -e .](https://github.com/aignacio/cocotbext-ahb/blob/master/noxfile.py#L30)) to normal [installation](https://github.com/aignacio/DELETE_ME/blob/master/noxfile.py#L38) otherwise the runner wasn't finding the namespace references. Am I missing something in my setup or should I avoid installing as editable during package development?

**Comment by ktbarrett** [comment]
Some of this code was updated in the 2.0 development cycle and it hasn't been fully tested. I anticipate additional changes, and it isn't an obvious thing to integrate into CI. So, there is no automated testing of this code; it might be broken on master right now.

The two environment variables you should need to set are `COCOTB_COVERAGE_RCFILE` and `COCOTB_USER_COVERAGE`. `COVERAGE_RCFILE` and `COVERAGE` are the old non-prefixed variable names that do the same thing.

As for the difference between editable and normal installation, I don't see how that could have change from something we did. Maybe it's due to a new version of coverage? Maybe there's a broken installation of your package in your site-packages directory? Are you running from the root of your project directory? If so Python could have be picking up the local and not the "installed" location.
